bndtools.core/src/bndtools/internal/BndAdapter.java

```diff
@@ -0,0 +1,42 @@
+package bndtools.internal;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IAdapterFactory;
+import org.osgi.service.component.annotations.Component;
+
+import aQute.bnd.build.Project;
+import bndtools.Plugin;
+import bndtools.central.Central;
+
+@Component(service = IAdapterFactory.class, property = {
+	IAdapterFactory.SERVICE_PROPERTY_ADAPTABLE_CLASS + "=org.eclipse.core.resources.IProject",
+	IAdapterFactory.SERVICE_PROPERTY_ADAPTER_NAMES + "=aQute.bnd.build.Project"
+})
+public class BndAdapter implements IAdapterFactory {
+
+	@Override
+	public <T> T getAdapter(Object adaptableObject, Class<T> adapterType) {
+		if (adapterType == Project.class) {
+			if (adaptableObject instanceof IProject project) {
+				try {
+					if (project.hasNature(Plugin.BNDTOOLS_NATURE)) {
+						// this project is for sure a bndtools project ...
+						Project bndproject = Central.getProject(project);
+						return adapterType.cast(project);
```

Comment:
Thanks that is fixed now, I sadly have had a bit of a hard time to setup a good testcase for that because bndtools was not able to create an inital (bnd-) workspace for me, that should now be fixed.
Weather or not Eclipse reports unused variables depend s on the eclipse project settings, I haven't checked the Bnd tools one ...